### PR TITLE
user docs: describe prerequisities to API methods

### PR DESCRIPTION
**Pull request by tobsan**
Adds a section under each method on the D-Bus API that describes what
prerequisities has to be fulfilled for a successful call to that method.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

### CODE REVIEW

File: doc/chapters/api/index.rst [hunk 1]
```diff
@@ -18,6 +18,13 @@ object path and interface.
 * Object Path: /com/pelagicore/SoftwareContainerAgent
 * Interface: com.pelagicore.SoftwareContainerAgent
 
+Prerequisities
+--------------
+The prerequisities for each method describe the absolute minimum needed for a call to that method to
```
Comment (kursatkobya) [inline doc/chapters/api/index.rst]: > the absolute minimum needed what actually ?

Comment (tobsan) [reply doc/chapters/api/index.rst]: The prerequisities is mostly that one has a container, but that is the minimum. It doesn't hinder using other API methods before calling the next one. Perhaps it makes little sense, in which case I should probably remove it.

File: doc/chapters/api/index.rst [hunk 2]
```diff
@@ -39,11 +46,16 @@ created.
...
 ##########
-* containerID: ``int32`` The ID obtained by CreateContainer method.
+* containerID: ``int32`` The ID obtained by Create method.
 * pathInHost: ``string`` absolute path in the host.
 * pathInContainer: ``string`` the absolute path to mount to in the container.
 * readOnly: ``bool`` indicates whether the directory is read-only or not.
 
+Prerequisities
+##############
+* A container that has been successfully created, such that the Create call returned a containerID.
```
Comment (kursatkobya) [inline doc/chapters/api/index.rst]: i would like more concise wording like : "Create method called and containerID is gathered"

File: doc/chapters/api/index.rst [hunk 3]
```diff
@@ -79,6 +91,10 @@ Return Values
 #############
 * containerID: ``int32`` ID of created SoftwareContainer.
 
+Prerequisities
+##############
+None
```
Comment (kursatkobya) [inline doc/chapters/api/index.rst:95]: can we mention about lxc dependency here ?

**Review by kursatkobya — APPROVED** [review]

File: doc/chapters/api/index.rst (hunk 3 above)
Comment (tobsan) [reply doc/chapters/api/index.rst:95]: There is no dependency for the user to have done anything before calling this method. To have LXC installed etc is not what these headings are about.

File: doc/chapters/api/index.rst (hunk 1 above)
Comment (thenor) [reply doc/chapters/api/index.rst]: I would rephrase it a bit.
"The prerequisites for each method specifies the mandatory parameters necessary for a successful method call"
Also I'm not a fan of the next sentence. I have to read it many times to understand it. Perhaps rephrase that also.
E.g. "For most calls the only prerequisite is a successfully created container, which is the same as getting a container ID returned from the Create method"
